File: cPP_java_py_js-DSA/data-structure/hashmap/hashmap.js
function printHashMap(mp){
    for(let [key,val] of mp ){
        console.log("key->"+key+"value->"+val );
    }
    return;
}

// map declaration
let mp = new Map();
console.log(mp);
printHashMap(mp);

//1. inserion operation : insert operation 
/*
    mp.set(key, value) : to set / insert some value 

    mp.get(key): to acess the value of key present in the map 

    mp.has(key): it check whether the key present in the map or not 
*/
mp[0]=1 ; 
mp[1]=10;
mp[20]=10;
printHashMap(mp);
console.log(mp[0]);

printHashMap(mp);

File: cPP_java_py_js-DSA/data-structure/hashmap/hashmap.cpp
#include<bits/stdc++.h>
using namespace std; 

void printHashMap(unordered_map<int ,int>mp ){
    for(auto it: mp ){
        cout<<"Key=" <<it.first<<",Value ="<<it.second<<"\n";
    }
    cout<<endl;
}
int main (){
     unordered_map<int,int>mp ; 


    // Operation in hashmap


    // 1.insert operation : mp[value] or insert method
    mp[0] =1 ;// key value insertion
    mp.insert({1,10});// using insert function 
    printHashMap(mp);


    // 2.acces value : mp[value] 
    cout<<mp[0]<<endl;// direct access

   

    //3. find method : give iterator 
     mp.find(3);
    printHashMap(mp);

    // 4. check if key exists 
    if(mp.find(3)!=mp.end()){
        cout<<"key exists ";
    }
    else{
        cout<<"Not exists ";
    }

    //5. erase method (iterator of key is passed)
    auto it= mp.find(0);
    mp.erase(it);
    printHashMap(mp);

    // 6. size method : to get the no of pairs exists in hashmap
    cout<<mp.size()<<endl; 

    // 7. clear method : to delete all the key value pairs from hashmap
    mp.clear(); 
    printHashMap(mp);

    // 8. iterating over hashmap: for (auto )loop

    for(auto it: mp ){
        cout<<"k->"<<it.first<<", value"<<it.second<<endl;
    }


    cout<<"how to use hash map and its operations"<<endl;
    return 0 ;

}